cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -134,24 +134,38 @@ public static LoggingContext getLoggingContext(String systemId, String component
 
   public static LoggingContext getLoggingContext(String namespaceId, String applicationId, String entityId,
                                                  ProgramType programType) {
-    return getLoggingContext(namespaceId, applicationId, entityId, programType, null);
+    return getLoggingContext(namespaceId, applicationId, entityId, programType, null, null);
   }
 
   public static LoggingContext getLoggingContextWithRunId(String namespaceId, String applicationId, String entityId,
-                                                          ProgramType programType, String runId) {
-    return getLoggingContext(namespaceId, applicationId, entityId, programType, runId);
+                                                          ProgramType programType, String runId,
+                                                          Map<String, String> systemArgs) {
+    return getLoggingContext(namespaceId, applicationId, entityId, programType, runId, systemArgs);
   }
 
   public static LoggingContext getLoggingContext(String namespaceId, String applicationId, String entityId,
-                                                 ProgramType programType, @Nullable String runId) {
+                                                 ProgramType programType, @Nullable String runId,
+                                                 @Nullable Map<String, String> systemArgs) {
     switch (programType) {
       case FLOW:
         return new FlowletLoggingContext(namespaceId, applicationId, entityId, "", runId, null);
       case WORKFLOW:
         return new WorkflowLoggingContext(namespaceId, applicationId, entityId, runId);
       case MAPREDUCE:
+        if (systemArgs != null && systemArgs.containsKey("workflowRunId")) {
```

Comment:
`RunRecordMeta.getSystemArgs` is marked `Nullable` probably because it will be null for the programs which are recorded before we started updating the runrecords with system args. Hence null check is required.